scraping_utils.py

```diff
@@ -293,4 +293,43 @@ def extract_timetables(string, year, term):
         out["courses"].append(course_out)
     return out
 
+def distribute_shared_sections(out):
+    messages = []
+    # For each course
+    for index, course in enumerate(out["courses"]):
+        course_code = course["subject_code"] + course["course_number"]
+        sec_comps = {}
+        for section in course["sections"]:
+            sec_comps[section["id"]] = []
+            for component in section["components"]:
+                if component["type"] not in sec_comps[section["id"]]:
+                    sec_comps[section["id"]].append(component["type"])
 
+        arr_lookup = lambda elt: len(sec_comps[elt["id"]])
+        comp_secs = group_by_eq(course["sections"], arr_lookup)
+
+        if len(comp_secs) >= 1:
+            continue
+
+        sections = []
+        bad_sec_ids = []
+        for section in course["sections"]:
+            if section["id"] in bad_sec_ids:
+                continue
+            for bad_section in comp_secs[1]:
+                if (len(sec_comps[section["id"]]) == 1) and (sec_comps[bad_section["id"]][0] == sec_comps[section["id"]][0]):
```

Comment:
I believe we're missing the check (at least for now) of `if sec_comps[bad_section["id"]][0] == "LEC": continue`.  Was there a reason this was omitted?
```suggestion
                if sec_comps[bad_section["id"]][0] == "LEC":
                    continue
                if (len(sec_comps[section["id"]]) == 1) and (sec_comps[bad_section["id"]][0] == sec_comps[section["id"]][0]):
```

Hopefully once non-open courses get scraped, that will eliminate the need for this ugly case.

Reply:
I think the code I based myself on wasn't totally up to date with the `develop` branch on the other repo, I guess I forgot to `git pull` or maybe I branched before you posted these changes.